osu.Game.Rulesets.Taiko/UI/TaikoPlayfieldAdjustmentContainer.cs

```diff
@@ -51,10 +55,38 @@ protected override void Update()
             // Limit the maximum relative height of the playfield to one-third of available area to avoid it masking out on extreme resolutions.
             relativeHeight = Math.Min(relativeHeight, 1f / 3f);
 
-            Y = base_position;
-
             Scale = new Vector2(Math.Max((Parent!.ChildSize.Y / 768f) * (relativeHeight / base_relative_height), 1f));
             Width = 1 / Scale.X;
         }
+
+        public double ComputeTimeRange()
+        {
+            float currentAspect = Parent!.ChildSize.X / Parent!.ChildSize.Y;
+
+            if (LockPlayfieldAspectRange.Value)
+                currentAspect = Math.Clamp(currentAspect, MINIMUM_ASPECT, MAXIMUM_ASPECT);
+
+            // in a game resolution of 1024x768, stable's scrolling system consists of objects being placed 600px (widthScaled - 40) away from their hit location.
+            // however, the point at which the object renders at the end of the screen is exactly x=640, but stable makes the object start moving from beyond the screen instead of the boundary point.
+            // therefore, in lazer we have to adjust the "in length" so that it's in a 640px->160px fashion before passing it down as a "time range".
+            // see stable's "in length": https://github.com/peppy/osu-stable-reference/blob/013c3010a9d495e3471a9c59518de17006f9ad89/osu!/GameplayElements/HitObjectManagerTaiko.cs#L168
+            const float stable_hit_location = 160f;
+            float widthScaled = currentAspect * stable_gamefield_height;
+            float inLength = widthScaled - stable_hit_location;
+
+            // also in a game resolution of 1024x768, stable makes hit objects scroll from 760px->160px at a duration of 6000ms, divided by slider velocity (i.e. at a rate of 0.1px/ms)
```

Comment:
where does 760px come from here?

Reply:
In a game resolution of 1024x768, the value `760px` comes from `160px + 600px` (where 160px is hit location, and 600px is stable's "in length").

In stable code, 760px is used specifically in [this line](https://github.com/peppy/osu-stable-reference/blob/013c3010a9d495e3471a9c59518de17006f9ad89/osu!/GameplayElements/HitObjectManagerTaiko.cs#L253) (`s.Position` is set to hit location [here](https://github.com/peppy/osu-stable-reference/blob/013c3010a9d495e3471a9c59518de17006f9ad89/osu!/GameplayElements/HitObjectManagerTaiko.cs#L201))